build.gradle.kts

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2019-2021 HRZN LTD
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import java.time.format.DateTimeFormatter
+import java.time.Year
+
+// Minecraft, Mappings, Loader Versions
+val minecraftVersion       = project.property("minecraft.version").toString()
+val yarnBuild              = project.property("yarn.build").toString()
+val loaderVersion          = project.property("loader.version").toString()
+
+// Mod Info
+val modVersion             = project.property("mod.version").toString()
+val modName                = project.property("mod.name").toString()
+val modGroup               = project.property("mod.group").toString()
+
+// Dependency Version
+val fabricVersion          = project.property("fabric.version").toString()
+val clothConfigVersion     = project.property("cloth.config.version").toString()
+val modMenuVersion         = project.property("modmenu.version").toString()
+val ucVersion              = project.property("uc.version").toString()
+val cardinalVersion        = project.property("cardinal.version").toString()
+val galacticraftApiVersion = project.property("galacticraft.api.version").toString()
+val reiVersion             = project.property("rei.version").toString()
+val cottonResourcesVersion = project.property("cotton.resources.version").toString()
+val fomlVersion            = project.property("foml.version").toString()
+val bannerppVersion        = project.property("bannerpp.version").toString()
+
+plugins {
+    id("fabric-loom") version("0.5-SNAPSHOT")
+    id("maven-publish")
+    id("org.cadixdev.licenser") version("0.5.0")
+    id("java")
+}
```

Comment:
Gradle recommends using statically-typed accessors for core-plugins since they are type-safe

```kotlin
plugins {
    java
    `maven-publish`
    id("fabric-loom") version "0.5-SNAPSHOT"
    id("org.cadixdev.licenser") version "0.5.0"
}
```